contracts/contracts/strategies/CurvePoolBooster.sol

```diff
@@ -0,0 +1,334 @@
+
+    /// @notice initialize function, to set up initial internal state
+    /// @param _strategist Address of the strategist
+    /// @param _fee Fee in BASE_FEE unit payed when managing campaign
+    /// @param _feeCollector Address of the fee collector
+    function initialize(
+        address _strategist,
+        uint16 _fee,
+        address _feeCollector
+    ) external onlyGovernor initializer {
+        _setStrategistAddr(_strategist);
+        _setFee(_fee);
+        _setFeeCollector(_feeCollector);
+    }
+
+    ////////////////////////////////////////////////////
+    /// --- MUTATIVE FUNCTIONS
+    ////////////////////////////////////////////////////
+    /// @notice Create a new campaign on VotemarketV2
+    /// @dev This will use all token available in this contract
+    /// @param numberOfPeriods Duration of the campaign in weeks
+    /// @param maxRewardPerVote Maximum reward per vote to distribute, to avoid overspending
+    /// @param blacklist  List of addresses to exclude from the campaign
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function createCampaign(
+        uint8 numberOfPeriods,
+        uint256 maxRewardPerVote,
+        address[] calldata blacklist,
+        uint256 bridgeFee,
+        uint256 additionalGasLimit
+    ) external onlyGovernorOrStrategist {
+        require(campaignId == 0, "Campaign already created");
+        require(numberOfPeriods > 1, "Invalid number of periods");
+        require(maxRewardPerVote > 0, "Invalid reward per vote");
+
+        // Handle fee (if any)
+        uint256 balanceSubFee = _handleFee();
+
+        // Approve the balanceSubFee to the campaign manager
+        IERC20(rewardToken).safeApprove(campaignRemoteManager, balanceSubFee);
+
+        // Create a new campaign
+        ICampaignRemoteManager(campaignRemoteManager).createCampaign{
+            value: bridgeFee
+        }(
+            ICampaignRemoteManager.CampaignCreationParams({
+                chainId: targetChainId,
+                gauge: gauge,
+                manager: address(this),
+                rewardToken: rewardToken,
+                numberOfPeriods: numberOfPeriods,
+                maxRewardPerVote: maxRewardPerVote,
+                totalRewardAmount: balanceSubFee,
+                addresses: blacklist,
+                hook: address(0),
+                isWhitelist: false
+            }),
+            targetChainId,
+            additionalGasLimit
+        );
+
+        emit BribeCreated(gauge, rewardToken, maxRewardPerVote, balanceSubFee);
+    }
+
+    /// @notice Manage the total reward amount of the campaign
+    /// @dev This function should be called after the campaign is created
+    /// @dev This will use all the token available in this contract
+    /// @param bridgeFee Fee to pay for the bridge
+    /// @param additionalGasLimit Additional gas limit for the bridge
+    function manageTotalRewardAmount(
```

Comment:
Also put `nonReentrant` on this one
(I know it is permissioned so highly unlikely to be re-entrency attacked)